ChangeSet
  1.1320.69.378 04/03/30 10:52:10 benh@kernel.crashing.org[torvalds] +1 -0
  [PATCH] ppc64: More incorrect syscall error test
  
  Oops, there was two different code path affected by this
  bug (strace and normal) and I fixed only one. Here's the
  other one:

  arch/ppc64/kernel/entry.S
    1.27.1.8 04/03/30 07:59:43 benh@kernel.crashing.org[torvalds] +1 -1
    ppc64: More incorrect syscall error test

diff -Nru a/arch/ppc64/kernel/entry.S b/arch/ppc64/kernel/entry.S
--- a/arch/ppc64/kernel/entry.S	Fri Apr  2 01:58:59 2004
+++ b/arch/ppc64/kernel/entry.S	Fri Apr  2 01:58:59 2004
@@ -194,7 +194,7 @@
 _GLOBAL(ret_from_syscall_2)
 	std	r3,RESULT(r1)	/* Save result */	
 	li	r10,-_LAST_ERRNO
-	cmpl	0,r3,r10
+	cmpld	0,r3,r10
 	blt	60f
 	neg	r3,r3
 57:	ld	r10,_CCR(r1)	/* Set SO bit in CR */
.........................................................................
# vim: syntax=diff

